pkg/nvlib/info/api.go

```diff
@@ -18,9 +18,15 @@ package info
 
 // Interface provides the API to the info package.
 type Interface interface {
+	Resolver
 	Properties
 }
 
+// Resolver defines a function to resolve a mode.
+type Resolver interface {
+	ResolvePlatform() Platform
+}
+
```

Comment:
Resolve a `mode` or resolve a `platform`?

Also, as a caller, `info.Resolver()` is not a very user-friendly name. maybe the more verbose `PlatformResolver`?

Reply:
Yes. This was initially copying the concept of a `mode` from the NVIDIA container toolkit, but it isn't flexible enought and needs to be a lower level to also support the device plugin use cases. The comment needs to be updated and I can update the type too.